Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Subscription Name: $($context.Subscription.Name)"
+    Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+    Write-Host "Account Name: $($context.Account.Id)"
+    Write-Host "Account Type: $($context.Account.Type)"
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "Checking if $($context.Account.Id) is allowed to run this script..."
+
+    # Checking if the current account type is "User"
+    if ($context.Account.Type -ne "User")
+    {
+        Write-Host "WARNING: This script can only be run by `User` Account Type. Account Type of $($context.Account.Id) is: $($context.Account.Type)" -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+
+    Write-Host "*** To disable HTTPS url scheme for API(s) of API Management services in a Subscription, Contributor and higher privileges on the API Management services are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 3] Preparing to fetch all API Management services..."
+    
+    if (-not (Test-Path -Path $FilePath))
+    {
+        Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+    Write-Host "Fetching all API Management services from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+    $validAPIManagementServiceDetails = @()
+    $apiManagementServiceDetails = Import-Csv -LiteralPath $FilePath
+    $apiManagementServiceDetails | ForEach-Object{
+        if( ![String]::IsNullOrWhiteSpace($_.ResourceId) -and ![String]::IsNullOrWhiteSpace($_.ResourceGroupName) -and ![String]::IsNullOrWhiteSpace($_.ResourceName) )
+        {
+            $validAPIManagementServiceDetails += $_
+        }
+    }
+
+    $totalAPIManagementResources = $($validAPIManagementServiceDetails.Count)
+
+    if ($totalAPIManagementResources -eq 0)
+    {
+        Write-Host "No API Management services found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalAPIManagementResources) API Management Service(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\DisableHTTPSUrlSchemeForAPIsOfAPIManagementServices"
```

Comment:
Please fix the file name `Url` vs `HTTPS`, and also to reflect the other file name related modifications being done in the remediation command.